File: EmbyStat.Web/ClientApp/src/app/shared/models/emby/emby-user.ts
import { Card } from '../common/card';
import { UserMediaView } from '../session/user-media-view';
import { EmbyUserAccessSchedule } from './emby-user-access-schedule';

export class EmbyUser {
  id: string;
  name: string;
  serverId: string;
  lastLoginDate: Date;
  lastActivityDate: Date;
  subtitleMode: string;
  isAdministrator: boolean;
  isHidden: boolean;
  isDisabled: boolean;
  maxParentalRating: number;
  accessSchedules: EmbyUserAccessSchedule[];
  invalidLoginAttemptCount: number;
  remoteClientBitRateLimit: number;
  deleted: boolean;
  primaryImageTag: string;
  viewedMovieCount: Card<number>;
  viewedEpisodeCount: Card<number>;
  lastWatchedMedia: UserMediaView[];
}

File: EmbyStat.Common/Models/Entities/EmbyUser.cs
namespace EmbyStat.Common.Models.Entities
{
    public class EmbyUser
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string ServerId { get; set; }
        public bool IsAdministrator { get; set; }
        public bool Deleted { get; set; }
    }
}
